include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -58,12 +58,22 @@ arena_prof_info_get(tsd_t *tsd, const void *ptr, alloc_ctx_t *alloc_ctx,
 		extent = iealloc(tsd_tsdn(tsd), ptr);
 	}
 
+	memset(prof_info, 0, sizeof(prof_info_t));
+
 	if (unlikely(!is_slab)) {
 		/* extent must have been initialized at this point. */
-		large_prof_info_get(extent, prof_info);
+		large_alloc_prof_info_get(extent, prof_info);
+
+		if ((uintptr_t)prof_info->alloc_tctx > (uintptr_t)1U) {
+			/*
+			 * This allocation was a prof sample.  Reset the
+			 * pointer on the recent allocation record, so that
+			 * this allocation is recorded as released.
```

Comment:
I might need to change the `prof_info_get()` name, but the reasoning here is: `prof_info_get()` is always called right before the allocation is freed, so we record that this allocation is freed in its corresponding record. It's possible to do so in some place other than `prof_info_get()`, but this is the most efficient place, because we've already tracked down to the `extent` associated with the allocation here; otherwise we'd need to track down the `extent` again.

Reply:
Yeah let’s either change the name to make it clear this is dealloc related, or split into two functions (even if they are always called together).

Reply:
Made the naming change as follows:
- `prof_info_get()` is a read-only operation;
- `prof_info_get_and_reset()` reads the `prof_info` and reset the associated recent allocation record.
- The underlying arena level getter accepts a boolean `reset` flag.